cdc/sink/dmlsink/txn/mysql/mysql.go

```diff
@@ -122,29 +127,28 @@ func NewMySQLBackends(
 
 	// Inherit the default value of the prepared statement cache from the SinkURI Options
 	cachePrepStmts := cfg.CachePrepStmts
-	prepStmtCacheSize := cfg.PrepStmtCacheSize
-
-	var stmtCache *lru.Cache
 	if cachePrepStmts {
 		// query the size of the prepared statement cache on serverside
 		maxPreparedStmtCount, err := pmysql.QueryMaxPreparedStmtCount(ctx, db)
 		if err != nil {
 			return nil, err
 		}
+		if maxPreparedStmtCount == -1 {
+			// NOTE: seems TiDB doesn't follow MySQL's specification.
+			maxPreparedStmtCount = math.MaxInt
```

Comment:
Should we change `math.MaxInt` to 1048576?

Reply:
I guess it's unnecessary. Even if in MySQL we still can set it bigger than 1048576.
So `-1` just means nolimit, so let's just keep it.